src/components/Extractor/DropDown.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useRef, useState, useEffect, RefObject } from "react"
+import Exp from "./Expression"
+import DownSelect from "./DownSelect"
+import { DropdownProps } from "./types"
+
+const Drop = (props: DropdownProps) => {
+  const {
+    placeholder,
+    EditorData,
+    node,
+    initialFocus = false,
+    options,
+    validationFn,
+    onChangeFn,
+    expressionRootClass,
+    expressionInputClass,
+  } = props
+
+  const [value, setValue] = useState<string>("")
+  const [exp, setExp] = useState<boolean>(false)
+  const dropRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (dropRef.current && initialFocus) {
+      dropRef.current.focus()
+    }
+  }, [dropRef.current])
+
+  const fnKeys: string[] = options
+    .filter(fn => fn.type === "function")
+    .map(fn => fn.key)
+
+  const handleValueChange = e => {
+    const val = e.target.value
+    // const newNode = new TreeNode(val)
+    node.setValue({ data: val, type: "string" })
+    setValue(val)
```

Comment:
Doesn't cause any problems since we're consuming them as strings while building the expression but better to be explicit about the type in the DS so sounds fair.